apps/api/src/persistence/repositories/__tests__/TestRepositoryProvider.ts

```diff
@@ -39,6 +42,26 @@ export default class TestRepositoryProvider extends RepositoryProvider {
         await Promise.all(writePromises);
     }
 
+    public async addCategories(categories: CategorizedTree): Promise<void> {
+        const categoryDocuments = categories
```

Comment:
Use the actual dto to doc mapping utility here.

Also, it may be time for a `cloneObjectAndRemoveProperties` util.